### ISSUE TITLE
Rule Errors do not propagate 

### ISSUE THREAD

**Issue by darkyen** [auth0/passport-auth0#28]
The following will not respect pass through any errors that come from Rules as `error_description`
we should probably just use the base .authenticate method instead.

```
Strategy.prototype.authenticate = function (req, options) {
  if (req.query && req.query.error) {
    return this.fail(req.query.error);
  }
  this._base.authenticate.call(this, req, options);
};
```

**Comment by techpool** [comment]
Yeah, actually what's happening is the fail function automatically redirects you to the failureRedirect path in the passport setting. I think this is a special case of auth0 and thus needs to be handled differently rather than calling the fail function.

**Comment by dshorowitz** [comment]
I have the same issue. Errors from rules get sent to the failureRedirect path. I want to show those errors in the lock widget instead (where the lock widget UI would update with the error text, similar to 'Wrong email or password'

![image](https://cloud.githubusercontent.com/assets/11761476/21867839/07cfda2e-d81e-11e6-9cdc-14e64a05017b.png)

How can you do this using auth0 passport?
